Objects/Objects/Geometry/Arc.cs

```diff
@@ -80,40 +92,38 @@ public Arc()
       this.endPoint = endPoint;
       this.angleRadians = angleRadians;
       this.applicationId = applicationId;
-      // TODO: 3D arcs
-      plane = new Plane(startPoint, new Vector(0, 0, 1), new Vector(1, 0, 0), new Vector(0, 1, 0), units);
 
       // find chord and chord angle which may differ from the arc angle
       var chordMidpoint = Point.Midpoint(startPoint, endPoint);
       var chordLength = Point.Distance(startPoint, endPoint);
       var chordAngle = angleRadians;
-      if ( chordAngle > Math.PI )
+      if (chordAngle > Math.PI)
         chordAngle -= Math.PI * 2;
-      else if ( chordAngle < -Math.PI )
+      else if (chordAngle < -Math.PI)
         chordAngle += Math.PI * 2;
       // use the law of cosines for an isosceles triangle to get the radius
       radius = chordLength / Math.Sqrt(2 - 2 * Math.Cos(chordAngle));
 
       // find the chord vector then calculate the perpendicular vector which points to the centre
       // which can be used to find the circle centre point
       var dir = chordAngle < 0 ? -1 : 1;
-      var centreToChord = Math.Sqrt(Math.Pow(( double )radius, 2) - Math.Pow(chordLength * 0.5, 2));
+      var centreToChord = Math.Sqrt(Math.Pow((double)radius, 2) - Math.Pow(chordLength * 0.5, 2));
       var perp = Vector.CrossProduct(new Vector(endPoint - startPoint), plane.normal);
       var circleCentre = chordMidpoint + new Point(perp.Unit() * centreToChord * -dir);
       plane.origin = circleCentre;
 
       // use the perpendicular vector in the other direction (from the centre to the arc) to find the arc midpoint
       midPoint = angleRadians > Math.PI
```

Comment:
note: this won't find the guaranteed midpoint for 180degree arcs

Reply:
This is the logic we already had, i only wrapped it into a constructor that took in a Plane to support 3D arcs, particularly for Breps, but in general too.

I agree that this logic is a bit flawed, I'll see if i can fix this quickly